src/Microsoft.AspNetCore.OData/Routing/ODataActionSelector.cs

```diff
@@ -95,10 +95,19 @@ public ActionDescriptor SelectBestCandidate(RouteContext context, IReadOnlyList<
         {
             RouteData routeData = context.RouteData;
             ODataPath odataPath = context.HttpContext.ODataFeature().Path;
+
             if (odataPath != null && routeData.Values.ContainsKey(ODataRouteConstants.Action))
             {
+                var odataRoute = routeData.Routers.OfType<ODataRoute>().FirstOrDefault();
+                var routePrefix = odataRoute?.RoutePrefix;
                 // Get the available parameter names from the route data. Ignore case of key names.
-                IList<string> availableKeys = routeData.Values.Keys.Select(k => k.ToLowerInvariant()).AsList();
+                // Remove route prefix and other non-parameter values from availableKeys
+                IList<string> availableKeys = routeData.Values.Keys
+                    .Where((key) => routePrefix != "{" + key + "}"
```

Comment:
>routePrefix != "{" + key + "} [](start = 36, length = 29)

what about if prefix has the same key name template?

Reply:
What do you mean by that?